clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -202,6 +202,11 @@ protected abstract void onJoinComplete(int generation,
                                            String protocol,
                                            ByteBuffer memberAssignment);
 
+    /**
+     * Invoked prior to each leave group event. This is typically used to cleanup assigned partitions
+     */
+    protected void onLeaveGroup() {}
```

Comment:
For consistency with `onJoinPrepare` and `onJoinComplete`, would it be reasonable to call this `onLeavePrepare`?

Reply:
Of course :)